### PR TITLE
Expanded upon Witch API. Fixes BUKKIT-3147

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/org/bukkit/entity/Witch.java [hunk 1]
```diff
@@ -4,4 +4,17 @@
...
+    /**
+     * Gets whether the witch is aggressive towards the player.
+     * 
+     * @return The witch's aggressiveness.
+     */
+    public boolean isAggressive();
+    
+    /**
+     * Sets whether the witch is aggressive towards the player.
+     *
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/Witch.java:16]: Spaces are bad.

File: src/main/java/org/bukkit/entity/Witch.java [hunk 2]
```diff
@@ -4,4 +4,17 @@
  * Represents a Witch
  */
 public interface Witch extends Monster {
+    /**
+     * Gets whether the witch is aggressive towards the player.
+     *
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/Witch.java:9]: Spaces are bad.

File: src/main/java/org/bukkit/entity/Witch.java [hunk 3]
```diff
@@ -4,4 +4,17 @@
  * Represents a Witch
  */
 public interface Witch extends Monster {
+    /**
+     * Gets whether the witch is aggressive towards the player.
+     * 
+     * @return The witch's aggressiveness.
+     */
+    public boolean isAggressive();
+
```
Comment (gmcferrin) [inline src/main/java/org/bukkit/entity/Witch.java:13]: Spaces are bad.

**Comment by amaranth** [comment]
Isn't this covered by EntityTrackEvent?

**Comment by riking** [comment]
You probably want to be cancelling EntityTargetEvent for witches, because the only thing you do here is add the ability to set whether they are aggressive.

**Comment by gmcferrin** [comment]
@amaranth 

> if EntityTargetEvent isn't fired for witches we should fix that instead

**Pull request closed without merging**